sdk/nodejs/tests/runtime/langhost/run.spec.ts

```diff
@@ -510,6 +514,41 @@ describe("rpc", () => {
                 });
             },
         },
+        "root_resource": {
+            program: path.join(base, "001.one_resource"),
+            expectResourceCount: 2,
+            showRootResourceRegistration: true,
+            registerResource: (ctx: any, dryrun: boolean, t: string, name: string, res: any, deps: string[],
+                               custom: boolean, protect: boolean, parent: string) => {
+                if (t === "pulumi:pulumi:Stack") {
+                    ctx.stackUrn = makeUrn(t, name);
+                    return { urn: makeUrn(t, name), id: undefined, props: undefined };
+                }
+
+                assert.strictEqual(t, "test:index:MyResource");
+                assert.strictEqual(name, "testResource1");
+                assert.strictEqual(parent, ctx.stackUrn);
+                return { urn: makeUrn(t, name), id: undefined, props: undefined };
+            },
+        },
+        "backcompat_root_resource": {
+            program: path.join(base, "001.one_resource"),
+            expectResourceCount: 2,
+            skipRootResourceEndpoints: true,
```

Comment:
If it's not clear here, this test deliberately does not provide implementations of `GetRootResource` and `SetRootResource`, which causes the Engine gRPC server to respond with the `UNIMPLEMENTED` status code - which is exactly what would happen if a user used this version of the NodeJS SDK with a Pulumi CLI that doesn't support this RPC endpoint.